### PR TITLE
Fix: Reduced Emojipicker landscape modal height

### PR DESCRIPTION
**Pull request by mananjadhav**
@Luke9389 Can you please review this?

### Details
- Reduced the EmojiPicker container height to allow closing picker

### Fixed Issues

$ #4541 

### Tests
1. Tested in landscape mode for all platforms
2. Tested portrait mode to see if it breaks any UI
3. Ensure the picker can be closed by clicking in the backdrop

### QA Steps
1. Go to any chat in Mobile Landscape mode
2. Click on Emoji Picker icon
3. Click on the area outside EmojiPicker to close it
4. 

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

#### Web
web-emojipicker

#### Mobile Web
mweb-landscape-emojipicker
mweb-portrait-emojipicker
https://user-images.githubusercontent.com/3069065/131915942-0ac84454-539b-4095-98d4-02c8f8a5eca1.mp4

#### Desktop
desktop-emojipicker

#### iOS
ios-landscape-emojipicker

#### Android
android-landscape-picker

### CODE REVIEW

File: src/components/withWindowDimensions.js [hunk 1]
```diff
@@ -46,11 +49,14 @@ export default function (WrappedComponent) {
             const isMediumScreenWidth = initialDimensions.width > variables.mobileResponsiveWidthBreakpoint
               && initialDimensions.width <= variables.tabletResponsiveWidthBreakpoint;
 
+            const isLandscape = (isMediumScreenWidth || isSmallScreenWidth) && initialDimensions.width >= initialDimensions.height;
```
Comment (marcaaron) [inline src/components/withWindowDimensions.js:52]: I get that checking if the width is greater than the height could tell us if we are in landscape. But why is the screen width relevant?

**Review by marcaaron — CHANGES_REQUESTED** [review]

File: src/components/withWindowDimensions.js (hunk 1 above)
Comment (mananjadhav) [reply src/components/withWindowDimensions.js:52]: Wanted to keep the flag enabled only for Tablets and Mobile by default. Will remove.

Comment (marcaaron) [reply src/components/withWindowDimensions.js]: Got it. If we want to do that does it make sense let's check the size of the screen wherever we are using this? 
Not sure I would expect an `isLandscape` prop to return `false` if we are past a certain width. We either have something oriented with its length shorter than width or we don't.

Comment (mananjadhav) [reply src/components/withWindowDimensions.js]: Noted. I've removed the screen width check. I guess `<Modal type={props.isSmallScreenWidth ? CONST.MODAL.MODAL_TYPE.BOTTOM_DOCKED : CONST.MODAL.MODAL_TYPE.POPOVER}` should cover for the screen width check that I need.

File: src/styles/getModalStyles.js [hunk 2]
```diff
@@ -4,7 +4,7 @@ import variables from './variables';
 import themeColors from './themes/default';
 
 export default (type, windowDimensions, popoverAnchorPosition = {}) => {
-    const {isSmallScreenWidth, windowWidth} = windowDimensions;
+    const {isSmallScreenWidth, windowWidth, isLandscape} = windowDimensions;
```
Comment (marcaaron) [inline src/styles/getModalStyles.js:7]: It seems premature to add `isLandscape`. We are only using this in one place and using `withWindowDimensions` in a zillion places. Could we alternatively just derive this by looking at the `windowWidth` and `windowHeight`?

Comment (mananjadhav) [reply src/styles/getModalStyles.js:7]: Okay sure. Do you see that in near future we won't be using it often? If yes, then I'll add the check here itself.

Comment (mananjadhav) [reply src/styles/getModalStyles.js]: Updated

Comment (marcaaron) [reply src/styles/getModalStyles.js]: That's the million dollar question. Our engineering culture at Expensify usually says **wait until we need the code** not **create with the expectation that something will be needed in the future**. Most of the time this attitude leads to less unnecessary complexity - and only sometimes leads to repetition. But it's easier to identify repetition than it is to identify things that are only used once.

**Review by marcaaron — APPROVED** [review]

File: src/styles/getModalStyles.js (hunk 2 above)
Comment (mananjadhav) [reply src/styles/getModalStyles.js]: Noted. Thanks for clarifying. Helpful for making better contributions 🙇

**Review by Luke9389 — APPROVED** [review]
OK great, looks good.

**Comment by botify** [comment]
This has been deployed to production and is now subject to a 7-day regression period.
If no regressions arise, payment will be issued on 2021-09-15.  :confetti_ball:

**Pull request merged**